### PR TITLE
chore(replace NATS): state ops enabled over grpc, tests run without nats

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- Move state ops from Node to Registry Service, enable state ops over grpc
- Remove deploying of NATS for tests from deployer.
- This PR potentially wraps up all usages of NATS

Signed-off-by: Abhinandan Purkait <abhinandan.purkait@mayadata.io>

### CODE REVIEW

**Comment by Abhinandan-Purkait** [comment]
bors try

File: control-plane/grpc/proto/v1/misc/common.proto [hunk 1]
```diff
@@ -97,6 +97,8 @@ enum ResourceKind {
   Watch = 11;
   // Spec
   Spec = 13;
+  // Spec
```
Comment (tiagolobocastro) [inline control-plane/grpc/proto/v1/misc/common.proto:100]: s/Spec/State

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/grpc/proto/v1/misc/common.proto (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/proto/v1/misc/common.proto]: Done.

File: deployer/src/infra/mod.rs [hunk 2]
```diff
@@ -422,7 +417,7 @@ impl_component! {
     Kibana,     1,
     Core,       3,
     JsonGrpc,   3,
-    Rest,       3,
+    Rest,       4,
```
[the remark this answers is not in the document]
Comment (Abhinandan-Purkait) [reply deployer/src/infra/mod.rs]: Reverted to 3

File: utils/deployer-cluster/src/lib.rs [hunk 3]
```diff
@@ -69,6 +69,8 @@ pub fn default_options() -> StartOptions {
         .with_io_engines(1)
         .with_show_info(true)
         .with_build_all(true)
+        // NATS will no longer be deployed for tests
+        .with_nats(false)
```
[the remark this answers is not in the document]
Comment (Abhinandan-Purkait) [reply utils/deployer-cluster/src/lib.rs]: Removed NATS part from deployer

File: control-plane/grpc/src/operations/registry/traits.rs [hunk 4]
```diff
@@ -102,3 +132,62 @@ impl From<message_bus::Specs> for registry::Specs {
...
     }
 }
+
+impl TryFrom<registry::States> for message_bus::States {
+    type Error = ReplyError;
+
+    fn try_from(value: registry::States) -> Result<Self, Self::Error> {
+        Ok(Self {
+            nexuses: {
+                let mut nexus_states: Vec<store::nexus::NexusState> = vec![];
```
[the remark this answers is not in the document]
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/registry/traits.rs:142]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Davincible** [comment]
Really looking forward to see this in the next release :)

**Pull request merged**